examples/checkout.c

```diff
@@ -0,0 +1,102 @@
+#include "common.h"
+#include <git2.h>
+#include <git2/checkout.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifndef _WIN32
+# include <pthread.h>
+# include <unistd.h>
+#endif
+
+
+typedef struct progress_data {
+    size_t completed_steps;
+    size_t total_steps;
+    const char *path;
+} progress_data;
+
+
+static void print_progress(const progress_data *pd)
+{
+    int checkout_percent = pd->total_steps
+                           ? (100 * pd->completed_steps / pd->total_steps)
+                           : 100;
+
+    printf( "chk %3d%% (%4" PRIuZ "/%4" PRIuZ ") %s\n",
+            checkout_percent,
+            pd->completed_steps, pd->total_steps,
+            pd->path );
```

Comment:
I feel like this function could easily be moved into its caller. Like this, you could even get rid of the payload completely, as all infos you need are already being provided to the callback